### PR TITLE
Added support for one liner pem certificates

### PR DESCRIPTION
**Pull request by Hakky54**

### CODE REVIEW

File: sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class PemFormatter {
+
+    private static final Pattern PEM_PATTERN = Pattern.compile("(-----.*?-----)(.*?)(-----.*?-----)");
```
Comment (GunnarPieter) [inline sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java:31]: What you could add is possible BEGIN and END patterns but this is also fine :)

File: sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        return Collections.emptyList();
+    }
+
+    private static String extractBase64EncodedString(String value, int startIndex, String innerEncryptionHeader) {
+        try {
+            String substring = value.substring(innerEncryptionHeader.length() + startIndex);
+            Base64.getDecoder().decode(substring);
+            return substring;
+        } catch (IllegalArgumentException exception) {
+            return extractBase64EncodedString(value, ++startIndex, innerEncryptionHeader);
```
Comment (GunnarPieter) [inline sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java:111]: If no correct value is found this loops until the string is out indexed. Might return a weird error?

**Review by GunnarPieter — APPROVED** [review]
Seems good to me :) Check my comments. Hope I helped a little

File: sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java (hunk 2 above)
Comment (Hakky54) [reply sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java]: This part was a work-in-progress when I needed to extract the inner body when an encryption header was present. I have now an alternative method, so this whole method is not needed any more, so I removed it.

File: sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java (hunk 1 above)
Comment (Hakky54) [reply sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java]: Thank you for the suggestion, I applied it!

File: sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java (hunk 2 above)
Comment (GunnarPieter) [reply sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java]: Nice, I also like the quick return in `extractInnerEncryptionHeaderIfPossible`. Java isn't so smart to initialize the hashmap after the potential quick return I think. But I could be wrong there.

Comment (Hakky54) [reply sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemFormatter.java]: Yeah, it is not so nice to recreate the map if it just only contains the those constants and never gets updates. It is a bit of a waste to recreate the map when the method is getting called. I refactored it to a constant now.

**Pull request merged**